drivers/usb/udc/udc_mcux_ehci.c

```diff
@@ -571,7 +571,7 @@ static int udc_mcux_ep_enable(const struct device *dev,
 	ep_init.zlt             = 0U;
 	ep_init.interval        = cfg->interval;
 	ep_init.endpointAddress = cfg->addr;
-	ep_init.maxPacketSize   = cfg->mps;
+	ep_init.maxPacketSize   = udc_mps_ep_size(cfg);
```

Comment:
In NXP MCUX controller driver, the `maxPacketSize` contains the bits of additional transactions. So it is right to use `cfg->mps`?

Reply:
If `ep_init.maxPacketSize` is the same as `wMaxPacketSize` then it should be fine to use `cfg->mps` directly. If the driver correctly supports high-bandwidth endpoints, then yes `high_bandwidth` capability must be set (otherwise stack will fail to initialize if SW requests high-bandwidth endpoint).

Reply:
HI @tmon-nordic , OK, Could you revert back the `maxPacketSize` changes of udc_mcux_ehci.c and udc_mdux_ip3511.c? I will create another pr to add the `high_bandwidth`. Is it OK?

Reply:
It was @jfischer-no requirement to cherry-pick his patch. I preferred to have this separate, because things are now piling up...

Can you just add the `high_bandwidth` (and remove the `udc_mps_ep_size()`) on top of this PR?

Reply:
Added. please help to check.

Reply:
You need to fix the signed-off-by in the commit.

Reply:
copy-paste issue. fixed.

Reply:
@tmon-nordic (I beg you, because it is probably faster) Please add a comment above the lines `ep_init.maxPacketSize   = cfg->mps;`, otherwise no one else will know this internal of the NXP HAL drivers.

Reply:
Not sure if wording is what you expected, but I went in with "HAL expects wMaxPacketSize value directly in maxPacketSize field".